src/common/cache_manager.cc

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2021-2025, XGBoost Contributors
+ */
+#include "cache_manager.h"
+#include "xgboost/logging.h"
+
+#include <cstdint>     // for uint64_t
+
+#if defined(__x86_64__)
+void RunCpuid(uint32_t eax, uint32_t ecx, uint32_t* abcd) {
```

Comment:
It would be safer in this variant:
```suggestion
void RunCpuid(uint32_t eax, uint32_t ecx, uint32_t (& abcd)[4]) {
```